src/main/kotlin/org/jetbrains/research/ml/codetracker/Plugin.kt

```diff
@@ -1,21 +1,107 @@
 package org.jetbrains.research.ml.codetracker
 
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
 import com.intellij.openapi.application.PathManager
+import com.intellij.openapi.application.ex.ApplicationEx
 import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import net.lingala.zip4j.ZipFile
+import java.io.File
+import java.io.InputStream
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardCopyOption
+
 
 enum class TestMode {
     ON,
     OFF
 }
 
+/**
+ * Represents a plugin required by Codetracker
+ */
+data class RequiredPlugin(val name: String, val zipFile: String) {
+    private val resourceFolder = "requiredplugins"
+    private val logger: Logger = Logger.getInstance(javaClass)
+
+    /**
+     * Perhaps there is a better way of doing this
+     */
+    fun isInstalled() : Boolean {
+        val isAlreadyInstalled =  File("${PathManager.getPluginsPath()}/${name}").exists()
+        logger.info("${Plugin.PLUGIN_ID}: plugin ${name} is already installed: $isAlreadyInstalled")
+        return isAlreadyInstalled
+    }
+
+    /**
+     * Not the best way for plugins installation but it works.
+     * Sure there is a way to install plugins properly (for example via PluginInstaller) but I have no time to find it out.
+     * Gets plugin .zip from resources, moves it to plugins dir and extracts.
+     */
+    fun install() : Boolean {
+        return if (!this.isInstalled()) {
+            val input: InputStream = javaClass.getResourceAsStream("$resourceFolder/$zipFile")
+            val targetPath = "${PathManager.getPluginsPath()}/${zipFile}"
+            Files.copy(input, Paths.get(targetPath), StandardCopyOption.REPLACE_EXISTING)
+            val zipFile = ZipFile(targetPath)
+            zipFile.extractAll(PathManager.getPluginsPath())
+            true
+        } else {
+            false
+        }
+    }
+}
+
 object Plugin {
     const val PLUGIN_ID = "codetracker"
     val testMode = TestMode.ON
     val codeTrackerFolderPath = "${PathManager.getPluginsPath()}/${PLUGIN_ID}"
 
+    private val requiredPlugins = arrayListOf(
+        RequiredPlugin("activity-tracker-plugin", "activity-tracker-plugin.zip"),
+        RequiredPlugin("JavaFX plugin", "JavaFX_plugin.zip")
```

Comment:
Is it a correct name? without hyphen?

Reply:
Yes, it is :) If you extract JavaFX zip file you'll see

Reply:
I have checked, it is correct